packages/core/src/RenderingEngine/VideoViewport.ts

```diff
@@ -445,6 +483,33 @@ class VideoViewport extends Viewport implements IVideoViewport {
     };
   }
 
+  public hasImageURI(imageURI: string) {
+    const testURI = imageURI.replace(/\/frames\/.*/, '');
+    if (this.imageId.indexOf(testURI) === -1) {
+      return false;
+    }
+    const frames = imageURI.match(/\/frames\/([^/?&]*)/);
+    if (!frames?.[1]) {
+      return true;
+    }
+    const regions = frames?.[1].split(',');
+    const frame = this.getFrame();
+    for (const region of regions) {
+      const regionSplit = region.split('-');
+      const minRegion = Number(regionSplit[0]);
+      const maxRegion =
+        regionSplit.length > 1 ? Number(regionSplit[1]) : minRegion;
+
+      if (minRegion === maxRegion && Math.abs(frame - minRegion) < 10) {
+        return true;
+      }
+      if (frame >= minRegion && frame <= maxRegion) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
can you please add some documentation on the logic here? what is regions, what is minRegion and maxRegion etc.

Reply:
What is this looking to? is the imageURI in the current range being played? in any of the frames?